numpy/lib/npyio.py

```diff
@@ -399,15 +399,12 @@ def load(file, mmap_mode=None, allow_pickle=True, fix_imports=True,
         pickle_kwargs = {}
 
     # TODO: Use contextlib.ExitStack once we drop Python 2
-    if isinstance(file, basestring):
-        fid = open(file, "rb")
-        own_fid = True
-    elif is_pathlib_path(file):
-        fid = file.open("rb")
-        own_fid = True
-    else:
+    if hasattr(file, 'read'):
         fid = file
         own_fid = False
```

Comment:
Some alternatives to this block:

This patch:
```python
if hasattr(file, 'read'):
    fid = file
    own_fid = False
else:
    fid = open(os_fspath(file), "rb")
    own_fid = True
```

Alternative 1 (smaller diff);
```python
if isinstance(file, os_PathLike):
    file = os_fspath(file)
if isinstance(file, basestring):
    fid = open(file, "rb")
    own_fid = True
else:
    fid = file
    own_fid = False
```

Alternative 2 (no duck-typing):
```python
if isinstance(file, io.IOBase):
    fid = file
    own_fid = False
else:
    fid = open(os_fspath(file), "rb")
    own_fid = True
```